### ISSUE TITLE
Request to better handle conflicts of interest in the judging team

### ISSUE THREAD

**Issue by 3docSec** [code-423n4/org#165]
TL;DR I believe we should do better to handle the case of conflict of interests for judges, that is the case when the judge has or has had in the past a business relation with the sponsor. I propose that a judge is disallowed to judge a contest where they have a business relation with the sponsor.

## The theory
Basing on the [Judge role definition on C4's website](https://docs.code4rena.com/roles/judges), I interpret judging as a well-needed impartial actor who makes the critical decision of validity/severity of issues.

In C4, there are unquestionable incentives:
1. for wardens to have their own findings found valid (make more money!)
2. for wardens to have other wardens' findings found not valid (make more money!)
3. for sponsors to have all wardens' findings found not valid or QA (who doesn't want to say "my code was found flawless"?)

The last point is also acknowledged in [the C4 docs about judging](https://docs.code4rena.com/roles/judges/how-to-judge-a-contest#heres-how-the-process-works-leading-up-to-judging):

> do note that sponsors are heavily biased against having a report that includes very many vulnerabilities. Focus your work as a judge on protecting users and providing feedback to wardens.

Because of all of the above, being truly independent is key to fair judging, and I would argue that a judge who has a business relationship with the sponsor can be more reluctant to openly contradict them, and therefore be in some way influenced by their heavy bias against having many vulnerabilities reported. This can be even more true if the judge audited the codebase themselves: any bugs found in the contest can be bugs they missed, so they themselves have good reasons for having a "cleaner" audit.

This point was discussed briefly in the past, and the current stance is if I got this right: 
- good SRs have a large network of clients and we don't want to keep the more senior judges from hopping in
- knowledge of the codebase is a plus that can be helpful to accurately judge

I personally don't find the above points compelling: like we prevent bias in asking the judge to not be compensated for their findings in the contests they judge, we should also address bias towards the sponsor, especially because for the time being only sponsors can appeal judges' decisions.

## The request

I would propose:
- that judges selected for a contest are required not to have, or have had a consulting relationship with the sponsor
- the above does NOT apply to validators, whose knowledge of the codebase can be very helpful

Or, at the very least, that contests where a conflict of interest exists, are open for wardens to invoke an appellate court.

**Comment by sockdrawermoney** [comment]
Thanks for raising. Going to take some time this week to consider and respond.

**Comment by sockdrawermoney** [comment]
I want to be clear that I'm sharing thoughts here and not describing any decisions or policies on the matter.

Regarding this:

> judges selected for a contest are required not to have, or have had a consulting relationship with the sponsor

At one point this was our goal, but the advice of the last supreme court session was to . Customers have preferences and I probably don't need to tell you that customers also have options.

> contests where a conflict of interest exists, are open for wardens to invoke an appellate court

Yes, I think this is how I see handling all cases where there are conflicts of interest. A judge is permitted to take on a scenario wherein they have a conflict of interest but are subject to greater scrutiny. Perhaps there is just a general appellate review budgeted by default out of the judge's share in cases where there is a conflict of interest—ultimately means that judging an audit where you have a conflict of interest makes you less money as a judge, which seems like a beneficial incentive anyway.

We have a conflict of interest disclosure process (Issue #1 in each findings repo) and I haven't looked at the degree to which we are successfully enforcing adherence to it but—given such a concern has never bubbled up to me in a conversation—I would guess it could be better.

---

PS For the avoidance of doubt, because of our relationship with Zellic, I want to be clear that there is currently no plan or discussion where Zellic would serve as judge. Independent judging is a hallmark of Code4rena and we intend to keep it so. In my view—and you may disagree with this—the cases where an independent judge has a conflict based on a relationship with an independent customer are a little less clearly cut-and-dry than the specific case of Zellic judging would be, where we currently have drawn a clear line.

**Comment by 0xEVom** [comment]
I do think warden-initiated appeals would go a long way towards addressing this, and may be easier to implement logistically than a pre-budgeted appellate court review. If there is an appellate review and anyone can request their finding to be reviewed (assuming that is how it would work), the risk I see is that there is no incentive not to escalate a finding. We can also assume judges will remain unbiased even in this situation most of the time, in which case there needs to be no appellate court if no one requests it.

Besides, warden-initiated appeals in all contests would also provide a way for wardens to mitigate bias from the judge in other situations such as:

- Negative/positive bias upon recognizing the warden that submitted a given finding
- Conflicts during PJQA
- Past experience or lack thereof with certain classes of vulnerabilities/protocols
- Sponsor influence, which can take place also outside of this situation

As high standard as judging is on C4, such mistakes are bound to happen on any platform as long as one person has the final decision-making power. And while PJQA is an effective tool in addressing technical errors in judging, it does not work as well to enforce lack of bias. This points to warden-initiated appeals having significant potential to further improve judging accuracy and impartiality.

I imagine some sponsors are allergic to the idea of further extending the judging phase (especially since there's little benefit for them in an appellate court review), even if by a small fraction of the total judging time. And also here, they have options. I don't know if this has been a factor in warden-initiated appeals not having been introduced yet, but I believe the argument of "most accurate judging of any platform" can also be quite powerful to attract both talented wardens and sponsors.

I would love to see warden-initiated appellate courts in all contests introduced for a period, and hopefully we can see a net positive effect.
